erpnext_germany/utils/eu_vat.py

```diff
@@ -1,12 +1,38 @@
-import zeep
+from zeep import Client
+import re
 
+WSDL_URL = "https://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl"
+COUNTRY_CODE_REGEX = r"^[A-Z]{2}$"
+VAT_NUMBER_REGEX = r"^[0-9A-Za-z\+\*\.]{2,12}$"
 
-def is_valid_eu_vat_id(vat_id) -> bool:
-	"""Use the EU VAT checker to validate a VAT ID."""
+
+def parse_vat_id(vat_id: str) -> tuple[str, str]:
 	country_code = vat_id[:2].upper()
 	vat_number = vat_id[2:].replace(" ", "")
 
-	client = zeep.Client("https://ec.europa.eu/taxation_customs/vies/checkVatService.wsdl")
-	result = client.service.checkVat(vatNumber=vat_number, countryCode=country_code)
+	# check vat_number and country_code with regex
+	if not re.match(COUNTRY_CODE_REGEX, country_code):
+		raise ValueError("Invalid country code")
+
+	if not re.match(VAT_NUMBER_REGEX, vat_number):
+		raise ValueError("Invalid VAT number")
+
+	return country_code, vat_number
+
+
+def check_vat(country_code: str, vat_number: str):
+	"""Use the EU VAT checker to validate a VAT ID."""
+	return Client(WSDL_URL).service.checkVat(
+		vatNumber=vat_number, countryCode=country_code
+	)
```

Comment:
Some exception handling would be good. One such case could be a typo in the country code of the VAT ID that raises an `INVALID_INPUT` Error from the Client or a an accidentally empty VAT ID sent in the request which garners the same error.

Reply:
Started by handling all the timeout and service unavailable errors with tenacity. Invalid input is still left to do.